core/storage/blog_statistics/gae_models.py

```diff
@@ -0,0 +1,842 @@
+                ' exists already.')
+        entity = cls(
+            id=author_id,
+            views_by_hour={},
+            views_by_date={},
+            views_by_month={}
+        )
+        entity.update_timestamps()
+        entity.put()
+
+        return entity
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id: str) -> bool:
+        """Check whether author blog post views stats model references user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.get_by_id(user_id) is not None
+
+
+class AuthorBlogPostAggregatedReadingTimeModel(base_models.BaseModel):
+    """A summary model to store total number of users staying for a particular
+    time on the blog posts written by a particular author. It will be updated as
+    soon as the user leaves the blog post written completely.
+    It will be keyed to the user ID of the author. This model can be
+    regenerated from 'BlogPostExitedEventLogEntryModel' in case of data
+    corruption.
+    """
+
+    # Number of user taking less than a minute to read the blog post.
+    zero_to_one_min = datastore_services.IntegerProperty(
+        indexed=False, required=True, repeated=False)
+    # Number of users taking one to two minutes to read the blog post.
+    one_to_two_min = datastore_services.IntegerProperty(
+        indexed=False, required=True, repeated=False)
+    # Number of users taking two to three minutes to read the blog post.
+    two_to_three_min = datastore_services.IntegerProperty(
+        indexed=False, required=True, repeated=False)
+    # Number of users taking three to four minutes to read the blog post.
+    three_to_four_min = datastore_services.IntegerProperty(
+        indexed=False, required=True, repeated=False)
+    # Number of users taking four to five minutes to read the blog post.
+    four_to_five_min = datastore_services.IntegerProperty(
+        indexed=False, required=True, repeated=False)
+    # Number of users taking five to six minutes to read the blog post.
+    five_to_six_min = datastore_services.IntegerProperty(
+        indexed=False, required=True, repeated=False)
+    # Number of users taking six to seven minutes to read the blog post.
+    six_to_seven_min = datastore_services.IntegerProperty(
+        indexed=False, required=True, repeated=False)
+    # Number of users taking seven to eight minutes to read the blog post.
+    seven_to_eight_min = datastore_services.IntegerProperty(
+        indexed=False, required=True, repeated=False)
+    # Number of users taking eight to nine minutes to read the blog post.
+    eight_to_nine_min = datastore_services.IntegerProperty(
+        indexed=False, required=True, repeated=False)
+    # Number of users taking nine to ten minutes to read the blog post.
```

Comment:
Are these comments needed? A single comment above this block should be okay?

Reply:
Removed, left only the neccesary ones.